Sources/GRPC/ClientResponseChannelHandler.swift

```diff
@@ -133,33 +141,55 @@ internal class ClientResponseChannelHandler<ResponseMessage: Message>: ChannelIn
   /// - status: the status promise is succeeded; if the status is not `ok` then the initial metadata
   ///   and response promise (if available) are failed with the status.
   public func channelRead(context: ChannelHandlerContext, data: NIOAny) {
-    guard self.inboundState != .ignore else { return }
+    guard self.inboundState != .ignore else {
+      self.logger.notice("ignoring read data: \(data)")
+      return
+    }
 
     switch self.unwrapInboundIn(data) {
     case .headers(let headers):
       guard self.inboundState == .expectingHeadersOrStatus else {
-        self.errorCaught(context: context, error: GRPCError.client(.invalidState("received headers while in state \(self.inboundState)")))
+        self.logger.error("invalid state '\(self.inboundState)' while processing headers")
```

Comment:
Should we log this error in `errorCaught` instead?

Reply:
We have less context in `errorCaught` (i.e. only the `Error`) so it might be better to keep it close to the source.